util/hot_backup.cc

```diff
@@ -0,0 +1,654 @@
+
+                    if (LiveOptions.env->FileExists(src_path))
+                        status=LiveOptions.env->RenameFile(src_path, dest_path);
+
+                    if (!status.ok())
+                    {
+                        good=false;
+                        Log(LiveOptions.info_log, "HotBackup failed while renaming %s slow directory",
+                            src_path.c_str());
+                    }   // if
+                }   // if
+            }
+            else
+            {
+                good=false;
+                Log(LiveOptions.info_log, "HotBackup failed while renaming %s directory",
+                    src_path.c_str());
+            }   // else
+        }   // for
+
+    }   // if
+    else
+    {
+        good=false;
+        Log(LiveOptions.info_log, "HotBackup failed while removing %s directory",
+            local_options.tiered_fast_prefix.c_str());
+    }   // else
+
+
+    // renaming old stuff succeeded, create new directory tree
+    if (good)
+    {
+        local_options=LiveOptions;
+        good=SetBackupPaths(local_options, 0);
+
+        status=LiveOptions.env->CreateDir(local_options.tiered_fast_prefix);
+        if (status.ok())
+        {
+            if (0 < local_options.tiered_slow_level)
+            {
+                status=LiveOptions.env->CreateDir(local_options.tiered_slow_prefix);
+                if (!status.ok())
+                {
+                    good=false;
+                    Log(LiveOptions.info_log, "HotBackup failed while creating %s slow directory",
+                        local_options.tiered_slow_prefix.c_str());
+                }   // if
+            }   // if
+        }   // if
+        else
+        {
+            good=false;
+            Log(LiveOptions.info_log, "HotBackup failed while creating %s directory",
+                local_options.tiered_fast_prefix.c_str());
+        }   // else
+
+        // now create all the sst_? directories
+        if (good)
+        {
+            status=MakeLevelDirectories(LiveOptions.env, local_options);
+            if (!status.ok())
+            {
+                good=false;
+                Log(LiveOptions.info_log, "HotBackup failed while creating sst_ directories");
+            }   // if
+        }   // if
+    }   // if
+
+    return(good);
+
+}   // DBImpl::PrepareDirectories
+
+
+/**
+ * Goal is to ensure the write buffer has not been sitting
+ * around for hours or days without a backup.  Not looking
+ * for perfect, up to the second backup of a constant input
+ * stream.
+ */
+bool
+DBImpl::PurgeWriteBuffer()
+{
+    bool good(true);
+    Status s;
+
+    // existing "test"
+    s=CompactMemTableSynchronous();
+    good=s.ok();
+
+    if (!good)
+        Log(GetLogger(), "HotBackup failed in PurgeWriteBuffer");
+
+    return(good);
+
+}   // DBImpl::PurgeWriteBuffer
+
+
+/**
+ * Need an independent MANIFEST and CURRENT file within
+ * hot backup directory.  This creates them.
```

Comment:
Recommend adding to this comment that this method also creates the file links. Could even change this method name to better indicate the tasks it performs; maybe something like WriteBackupManifestAndLinkFiles().